### PR TITLE
docs: added NatSpec comments for borrower contracts

### PR DESCRIPTION
**Pull request by xmxanuel**
I started to add `NatSpec` comments for the borrower contracts.
It is a commonly used standard nowadays for Solidity.

Writing those comments is surprisingly easy together with Github Copilot. 

For me, it is also an excellent way to refresh my understanding of the codebase again.

### CODE REVIEW

File: src/borrower/pile.sol [hunk 1]
```diff
@@ -123,33 +161,14 @@ contract Pile is Auth, Interest {
...
-            rates[rate].fixedRate = value;
-        } else {
-            revert("unknown parameter");
-        }
-
-        emit File(what, rate, value);
-    }
-
-    // accrue needs to be called before any debt amounts are modified by an external component
+    /// @notice accrue needs to be called before any debt amounts are modified by an external component
```
Comment (AStox) [inline src/borrower/pile.sol:164]:
```suggestion
    /// @notice calls drip on the given loan
    /// @dev accrue needs to be called before any debt amounts are modified by an external component
```

File: src/borrower/shelf.sol [hunk 2]
```diff
@@ -179,7 +192,10 @@ contract Shelf is Auth, TitleOwned, Math {
         emit Borrow(loan, currencyAmount);
     }
 
-    // withdraw transfers the currency to the borrower account
+    /// @notice withdraw transfers the acutal currency to the borrower account
```
Comment (AStox) [inline src/borrower/shelf.sol:195]:
```suggestion
    /// @notice withdraw transfers the actual currency to the borrower account
```

File: src/borrower/shelf.sol [hunk 3]
```diff
@@ -236,25 +257,9 @@ contract Shelf is Auth, TitleOwned, Math {
...
-
-        reserve.deposit(currencyAmount);
-        // reBalance lender interest bearing amount based on new NAV
-        assessor.reBalance();
-    }
-
-    // locks an nft in the shelf
-    // requires an issued loan
+    /// @notice locks an nft in the shelf
+    /// requires an issued loan
```
Comment (AStox) [inline src/borrower/shelf.sol:261]:
```suggestion
    /// @dev requires an issued loan
```

File: src/borrower/shelf.sol [hunk 4]
```diff
@@ -263,8 +268,9 @@ contract Shelf is Auth, TitleOwned, Math {
         emit Lock(loan);
     }
 
-    // unlocks an nft in the shelf
-    // requires zero debt or 100% write off
+    /// @notice unlocks an nft in the shelf
+    /// requires zero debt or 100% write off
```
Comment (AStox) [inline src/borrower/shelf.sol:272]:
```suggestion
    /// @dev requires zero debt or 100% write off
```

File: src/borrower/wrappers/writeOffWrapper.sol [hunk 5]
```diff
@@ -48,19 +50,22 @@ contract WriteOffWrapper is Auth, Discounting {
         writeOffRates[address(0x4b0f712Aa9F91359f48D8628De8483B04530751a)] = 1001; // Peoples 1
     }
 
-    function writeOff(uint256 _loanID, address _feed) public auth {
-        FeedLike feed = FeedLike(_feed);
+    /// @notice writes of an overdue loan
```
Comment (AStox) [inline src/borrower/wrappers/writeOffWrapper.sol:53]:
```suggestion
    /// @notice writes off an overdue loan
```

**Review by AStox — CHANGES_REQUESTED** [review]
Just a few more typos and nit picks

**Review by AStox — DISMISSED** [review]
Looks great!

**Review by offerijns — APPROVED** [review]

**Review by AStox — APPROVED** [review]
LGTM

**Pull request merged**